### PR TITLE
Feature/76/dynatraceslosupport

### PR DESCRIPTION
**Pull request by grabnerandi**
work in progress

### CODE REVIEW

**Comment by grabnerandi** [comment]
closing #112 - you can now have SLIs on every level (service, stage & project) and they are correctly "merged/inherited" as described in the doc
closing #109 - Dynatrace API Responses are now better handled. When unexpected errors happen they will be logged correctly instead of resulting in a parsing error
closing #97 - Now supporting native Dynatrace SLO definitions for both dashboard parsing as well as introducing a query option in sli.yaml
closing #104  - will now include better logging which includes the actual query string that was sent to the dynatrace metrics api
closing #105 - will now produce a logoutput in case we default to one of the 4 core SLIs (error_rate, p90, ...)

File: cmd/main.go [hunk 1]
```diff
@@ -218,14 +290,15 @@ func retrieveMetrics(event cloudevents.Event) error {
...
 
-	//
-	// Lets get a new Keptn Handler
-	keptnHandler, err := keptn.NewKeptn(&event, keptn.KeptnOpts{
-		ConfigurationServiceURL: os.Getenv(configservice),
-	})
-	if err != nil {
-		return err
-	}
+	/* 	//
```
Comment (bacherfl) [inline cmd/main.go]: let's delete those lines instead of commenting them out

File: cmd/main.go [hunk 2]
```diff
@@ -203,6 +247,34 @@ func getDataFromDynatraceDashboard(dynatraceHandler *dynatrace.Handler, keptnEve
...
+ */
+func getDynatraceProblemContext(eventData *keptn.InternalGetSLIEventData) string {
+
+	// iterate through the labels and find Problem URL
+	if eventData.Labels == nil || len(eventData.Labels) == 0 {
+		return ""
+	}
+
+	for labelName, labelValue := range eventData.Labels {
+		if labelName == "Problem URL" {
```
Comment (bacherfl) [inline cmd/main.go]: to be safe, i suggest to convert both strings to lower case, e.g. `if strings.ToLower(labelName) == "problem url"

File: README.md [hunk 3]
```diff
@@ -39,7 +45,6 @@ As *dynatrace-sli-service* uses the Metrics API v2 this opens up your SLIs to an
 |   0.6.1,0.6.2    | keptncontrib/dynatrace-sli-service:0.4.2 |
 |   0.7.0    | keptncontrib/dynatrace-sli-service:0.5.0 |
 |   0.7.1    | keptncontrib/dynatrace-sli-service:0.6.0 |
-|   0.7.2    | keptncontrib/dynatrace-sli-service:0.7.0 |
```
Comment (bacherfl) [inline README.md:42]: We should keep this line and add a new line with  `0.7.2 -> dynatrace-sli-service:0.7.1`

File: README.md [hunk 4]
```diff
@@ -24,10 +24,16 @@ By default these metrics (SLIs) are queried from a Dynatrace-monitored service e
 
 As highlighted above, the *dynatrace-sli-service* also provides the following capabilities
 * Connecting to different Dynatrace Tenants (SaaS or Managed) depending on Keptn Project, Stage or Service
-* Defining a custom list of SLIs based on the Dynatrace Metrics API v2 in your SLI.yaml
+* Defining a custom list of SLIs based on the Dynatrace Metrics API v2. As the *The following is an example from above:
+
+
+dynatrace-sli-provider*  in your SLI.yamlexecutes your query
```
Comment (bacherfl) [inline README.md]: typo: `SLI.yamlexecutes`

File: pkg/common/common.go [hunk 5]
```diff
@@ -203,6 +254,92 @@ func GetKeptnResource(keptnEvent *BaseKeptnEvent, resourceURI string, logger *ke
...
+	if sliFilePath != "" {
+		localFileContent, err := ioutil.ReadFile(sliFilePath)
+		if err != nil {
+			logMessage := fmt.Sprintf("Couldn't load file content from %s", sliFilePath)
+			logger.Info(logMessage)
+			return nil, nil
+		}
+		logger.Info("Loaded LOCAL file " + sliFilePath)
+		sliFileContent = string(localFileContent)
+	} else {
```
Comment (bacherfl) [inline pkg/common/common.go]: this `else {}` block can be removed

File: pkg/common/common.go [hunk 6]
```diff
@@ -203,6 +254,92 @@ func GetKeptnResource(keptnEvent *BaseKeptnEvent, resourceURI string, logger *ke
...
+
+	// couldnt load any SLIs
+	if len(sliMap) == 0 {
+		logger.Info(fmt.Sprintf("No custom SLI queries for project=%s,stage=%s,service=%s found as no dynatrace/sli.yaml in repo. Going with default!", keptnEvent.Project, keptnEvent.Stage, keptnEvent.Service))
+	} else {
+		logger.Info(fmt.Sprintf("Found a total of %d SLI queries for project=%s,stage=%s,service=%s in dynatrace/sli.yaml in locations: %s", len(sliMap), keptnEvent.Project, keptnEvent.Stage, keptnEvent.Service, foundLocation))
+	}
+
+	// load dynatrace/sli.yaml - if its there we add it to the sliMap
+	/* if err != nil {
```
Comment (bacherfl) [inline pkg/common/common.go]: let's remove this code instead of commenting it out

**Review by bacherfl — CHANGES_REQUESTED** [review]
Overall the PR looks good! I just have a few minor comments

**Review by bacherfl — APPROVED** [review]
lgtm

**Pull request merged**